app/services/questions_service.rb

```diff
@@ -46,28 +46,39 @@ def add_application_level_tasks_to_response(proceeding_type)
     applicable_tasks = proceeding_type.application_tasks
     applicable_tasks.each do |task|
       display_rules = task.proceeding_type_merits_tasks.find_by(proceeding_type_id: proceeding_type.id).display_rules
-      display_task = display_rules.nil? || send(display_rules)
+      display_task = display_rules.nil? || send(display_rules, proceeding_type)
       @response[:application][:tasks][task.name] = task.dependency_names if display_task
     end
   end
 
   def add_proceeding_level_tasks_to_response(proceeding_type, ccms_code)
     pt_hash = { ccms_code:, tasks: {} }
     proceeding_type.proceeding_tasks.each do |task|
-      pt_hash[:tasks][task.name] = task.dependency_names
+      display_rules = task.proceeding_type_merits_tasks.find_by(proceeding_type_id: proceeding_type.id).display_rules
+      display_task = display_rules.nil? || send(display_rules, proceeding_type)
```

Comment:
**not a blocker** just thinking whether its worth extracting to a private method as is a. pretty dense, b. looks to be used at line 48 and 49 above too. 

```
def display_task
  display_rules = task.proceeding_type_merits_tasks.find_by(proceeding_type_id: proceeding_type.id).display_rules
  display_task = display_rules.nil? || send(display_rules, proceeding_type)
end
```